packages/viem/src/index.ts

```diff
@@ -0,0 +1,291 @@
+ */
+type TApiKeyAccountConfig = {
+  /**
+   * Turnkey API public key
+   */
+  apiPublicKey: string;
+  /**
+   * Turnkey API private key
+   */
+  apiPrivateKey: string;
+  /**
+   * Turnkey API base URL
+   */
+  baseUrl: string;
+  /**
+   * Turnkey organization ID
+   */
+  organizationId: string;
+  /**
+   * Turnkey private key ID
+   */
+  privateKeyId: string;
+};
+
+/**
+ * Creates a new Custom Account backed by a Turnkey API key.
+ */
+export async function createApiKeyAccount(
+  config: TApiKeyAccountConfig
+): Promise<LocalAccount> {
+  const { apiPublicKey, apiPrivateKey, baseUrl, organizationId, privateKeyId } =
+    config;
+
+  httpInit({
+    apiPublicKey,
+    apiPrivateKey,
+    baseUrl,
+  });
+
+  const data = await TurnkeyApi.getPrivateKey({
+    body: {
+      privateKeyId: privateKeyId,
+      organizationId: organizationId,
+    },
+  });
+
+  const ethereumAddress = data.privateKey.addresses.find(
+    (item: any) => item.format === "ADDRESS_FORMAT_ETHEREUM"
+  )?.address;
+
+  if (typeof ethereumAddress !== "string" || !ethereumAddress) {
+    throw new TurnkeyActivityError({
+      message: `Unable to find Ethereum address for key ${privateKeyId} under organization ${organizationId}`,
+    });
+  }
+
+  return toAccount({
+    address: ethereumAddress as `0x${string}`,
+    signMessage: function ({
+      message,
+    }: {
+      message: SignableMessage;
+    }): Promise<`0x${string}`> {
+      return signMessage(message, organizationId, privateKeyId);
+    },
+    signTransaction: function <
+      TTransactionSerializable extends TransactionSerializable
+    >(
+      transaction: TTransactionSerializable,
+      args?:
+        | { serializer?: SerializeTransactionFn<TTransactionSerializable> }
+        | undefined
+    ): Promise<`0x${string}`> {
+      const serializer = !args?.serializer
+        ? serializeTransaction
+        : args.serializer;
+
+      return signTransaction(
+        transaction,
+        serializer,
+        organizationId,
+        privateKeyId
+      );
+    },
+    signTypedData: function (
+      typedData: TypedData | { [key: string]: unknown }
+    ): Promise<`0x${string}`> {
+      return signTypedData(typedData, organizationId, privateKeyId);
+    },
+  });
+}
+
+async function signMessage(
+  message: SignableMessage,
+  organizationId: string,
+  privateKeyId: string
+): Promise<`0x${string}`> {
+  const hashedMessage = keccak256(message as `0x${string}`);
+  const signedMessage = await signMessageWithErrorWrapping(
+    hashedMessage,
+    organizationId,
+    privateKeyId
+  );
+  return `${signedMessage}` as `0x${string}`;
+}
+
+async function signTransaction<
+  TTransactionSerializable extends TransactionSerializable
+>(
+  transaction: TTransactionSerializable,
+  serializer: SerializeTransactionFn<TTransactionSerializable>,
+  organizationId: string,
+  privateKeyId: string
+): Promise<`0x${string}`> {
+  const serializedTx = serializer(transaction);
```

Comment:
It looks like these serializers may be user supplied. If thats the case, are we sure that these will produce a format that our API understands? Is it worth it to do some validation on the payload itself clientside if thats the case. If my assumption is wrong, please ignore!